Add details about the full release lifecycle (#36954)
dotnet/docs: docs/fundamentals/package-validation/baseline-version-validator.md

However, when you try to pack, it throws an error.

> ```text

Reviewer: ```suggestion
```dotnetcli
```

D:\demo>dotnet pack
MSBuild version 17.3.2+561848881 for .NET
  Determining projects to restore...
  All projects are up-to-date for restore.
  AdventureWorks.Client -> D:\demo\bin\Debug\net6.0\AdventureWorks.Client.dll
C:\Program Files\dotnet\sdk\6.0.413\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Compatibility.Common.targets(33,5): error CP0002: Member 'A.B.Connect(string)' exists on [Baseline] lib/net6.0/AdventureWorks.Client.dll but not on lib/net6.0/AdventureWorks.Client.dll [D:\demo\AdventureWorks.Client.csproj]
```

> ![BaselineVersion](media/baseline-version.png)

Author: @ViktorHofer @IEvangelist I have now also updated the screenshots

Now when you pack the project, it succeeds.

![BaselineVersionSuccessful](media/baseline-version-successful.png)

For version 2.0.0 you later decide you want to remove obsolete `Connect` method with the single `string` parameter and upon careful consideration deliberately decide to accept this breaking change.

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net6.0</TargetFramework>
    <PackageVersion>2.0.0</PackageVersion>
    <EnablePackageValidation>true</EnablePackageValidation>
    <PackageValidationBaselineVersion>1.1.0</PackageValidationBaselineVersion>
  </PropertyGroup>

</Project>
```

```diff
- public static HttpClient Connect(string url)
- {
-     return Connect(url, Timeout.InfiniteTimeSpan);
- }

public static HttpClient Connect(string url, TimeSpan timeout)
{
    // ...
}
```

> To suppress the `CP0002` error for this intentional breaking change, you can add a *CompatibilitySuppressions.xml* file to your project.
> By calling `dotnet pack /p:GenerateCompatibilitySuppressionFile=true` once, this file can be generated automatically.  
> It will contain a suppression for each validation error that occurred during pack (see also [Suppress compatibility errors](overview.md#suppress-compatibility-errors)).

Reviewer: ```suggestion
To suppress the `CP0002` error for this intentional breaking change, you can add a *CompatibilitySuppressions.xml* file to your project. By calling `dotnet pack /p:GenerateCompatibilitySuppressionFile=true` once, this file can be generated automatically. It contains a suppression for each validation error that occurred during pack. For more information, see [Suppress compatibility errors](overview.md#suppress-compatibility-errors).
```

> In our case *CompatibilitySuppressions.xml* will contain the suppression for the `CP0002` error:

Reviewer: ```suggestion
In this example, the *CompatibilitySuppressions.xml* contains the suppression for the `CP0002` error:
```

  <PropertyGroup>
    <TargetFramework>net6.0</TargetFramework>
    <PackageVersion>2.0.0</PackageVersion>
    <EnablePackageValidation>true</EnablePackageValidation>
    <PackageValidationBaselineVersion>1.1.0</PackageValidationBaselineVersion>
    <GenerateCompatibilitySuppressionFile>true</GenerateCompatibilitySuppressionFile>  
  </PropertyGroup>

</Project>
```

If `GenerateCompatibilitySuppressionFile` is set to `true`, all validation errors that occur during pack will be added as a suppression to the generated `CompatibilitySuppressions.xml` file.

> This file could be checked into source control to document and review the breaking changes made in a PR and the upcoming release.

Reviewer 2: First of all, really great additions so far. Thank you for that.

The `GenerateCompatibilitySuppressionFile` property is intended to be passed in once, to generate or update the suppression file. It's not meant to be a property that you permanently set in your project file and then is true for all continuous builds. If you do so, you would accept other breaking as well by automatically getting them added to the suppression file.

Instead, the common case (or better to say, the case that we designed this for) is to set the `GenerateCompatibilitySuppressionFile=true` property just once. When working on the CLI you would do that via the following command: `dotnet pack /p:GenerateCompatibilitySuppressionFile=true`. Inside Visual Studio there isn't a great way to do this without creating distinct launch profiles (same for vscode). So you could set the property in the project file, then invoke `Pack` and then after the suppression file is generated / updated, remove the property from the project file.

> This file could be checked into source control to document and review the breaking changes made in a PR and the upcoming release.

Reviewer 2: Agreed, we should document the format.

> This file could be checked into source control to document and review the breaking changes made in a PR and the upcoming release.

Reviewer 2: The word _should_ communicates the need to preserve that file in the tree.

```suggestion
This file should be checked into source control to document and review the breaking changes made in a PR and the upcoming release.
```
